geoindicators/src/main/groovy/org/orbisgis/RsuIndicators.groovy

```diff
@@ -619,4 +615,164 @@ static IProcess rsuEffectiveTerrainRoughnessHeight() {
+                        }
+                        filtering = filtering[0..-4]
+                    }
+                    def selectionQuery = "DROP TABLE IF EXISTS $outputTableName; DROP TABLE IF EXISTS $roadInter; " +
+                            "CREATE INDEX IF NOT EXISTS ids_r ON $roadTable($geometricColumnRoad) USING RTREE; " +
+                            "CREATE INDEX IF NOT EXISTS ids_u ON $rsuTable($geometricColumnRsu) USING RTREE; " +
+                            "CREATE TABLE $roadInter AS SELECT a.$idColumnRsu AS id_rsu, " +
+                            "ST_AREA(a.$geometricColumnRsu) AS rsu_area, ST_INTERSECTION(a.$geometricColumnRsu, " +
+                            "b.$geometricColumnRoad) AS the_geom $ifZindex FROM $rsuTable a, $roadTable b " +
+                            "WHERE $filtering;"
+                    datasource.execute(selectionQuery.toString())
+
+                    // If all roads are considered at the same level...
+                    if(levelConsiderated==null){
+                        nameDens.add("rsu_linear_road_density")
+                        caseQueryDens = "SUM(ST_LENGTH(the_geom))/rsu_area AS rsu_linear_road_density "
+                        for (int d=angleRangeSize; d<=180; d+=angleRangeSize){
+                            caseQueryDistrib += "SUM(CASEWHEN(azimuth>=${d-angleRangeSize} AND azimuth<$d, length, 0)) AS " +
+                                    "rsu_road_direction_distribution_d${d-angleRangeSize}_$d,"
+                        }
+                    }
+                    // If only certain levels are considered independantly
+                    else{
+                        ifZindex = ", zindex "
+                        levelConsiderated.each { lev ->
+                            caseQueryDens += "SUM(CASEWHEN(zindex = $lev, ST_LENGTH(the_geom), 0))/rsu_area " +
+                                    "AS rsu_linear_road_density_h${lev.toString().replaceAll('-','minus')},"
+                            nameDens.add("rsu_linear_road_density_h${lev.toString().replaceAll('-','minus')}")
+                            for (int d=angleRangeSize; d<=180; d+=angleRangeSize){
+                                caseQueryDistrib += "SUM(CASEWHEN(azimuth>=${d-angleRangeSize} AND azimuth<$d AND " +
+                                        "zindex = $lev, length, 0)) AS " +
+                                        "rsu_road_direction_distribution_h${lev.toString().replaceAll("-","minus")}" +
+                                        "_d${d-angleRangeSize}_$d,"
+                            }
+                        }
+                    }
+
+                    // QUERIES EXECUTED DEPENDING ON THE CONTENT OF THE OPERATION LIST
+                    // If the road direction distribution is calculated, explode the roads into segments in order to calculate
+                    // their length for each azimuth range
+                    if (operations.contains("rsu_road_direction_distribution")) {
```

Comment:
what about if operations is null or empty ?

Reply:
Good question. It is also a problem if we have a String not defined in the operations. This is true for this indicator but also for several others that we have already coded (for example the first indicator coded "geometryProperties"). Should we return an error ?

Reply:
If the operation name is not conform or null we must log an error.
@SPalominos

Reply:
which solution is chosen ?

Reply:
I have added several tests to log several kind of errors. A problem still exists if the user set the operations as a String or a number instead of a list of... If you validate the way I log the error, we should add an issue to correct all geoindicators that use the same kind of inputs (operations) at the end.